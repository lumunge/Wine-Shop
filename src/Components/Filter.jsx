import React, { Component } from 'react'

export default class Filter extends Component {
    render() {
        return (
            <div className="filter-component">
                    <div className="filter-content">
                        <div className="order mb-4">
                            <div>
                                Show me {" "}
                            </div>
                            <select value={this.props.variety} onChange={this.props.filterWines} name="" id="">
                                <option value="">All</option>
                                <option value="white">White</option>
                                <option value="red">Red</option>
                                <option value="sparkling">Sparkling</option>
                            </select>
                        </div>
                        <div className="order">
                            <div className="">
                                Order by {" "}
                            </div>
                            <select value={this.props.sort} onChange={this.props.sortWines} name="" id="">
                                <option value="">Order by</option>
                                <option value="bottle-price">Price</option>
                                <option value="case-price">Vintage</option>
                            </select>
                        </div>
                    </div>
                </div>
        )
    }
}
