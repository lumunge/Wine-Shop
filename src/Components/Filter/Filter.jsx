import React, { Component } from 'react'

export default class Filter extends Component {
    render() {
        return (
            <div className="filter-component">
                    <div className="filter-content">
                        <div className="order mb-4">
                            <div className="filter-header">
                                <span>Show Me</span>
                                <span>Show All</span>
                            </div>
                            <div className="filter-buttons">
                                <button data-testid="white" className="filter-btn" value="white" onClick={this.props.filterWines}>White</button>
                                <button data-testid="red" className="filter-btn" value="red" onClick={this.props.filterWines}>Red</button>
                                <button data-testid="sparkling" className="filter-btn" value="sparkling" onClick={this.props.filterWines}>Sparkling</button>
                            </div>
                        </div>
                        <div className="order">
                            <div>
                                <span> Order by {" "} </span>
                                <button className="filter-btn" value="bottle-price" onClick={this.props.sortWines}>Price</button>
                                <button className="filter-btn" value="case-price" onClick={this.props.sortWines}>Vintage</button>
                            </div>
                        </div>
                    </div>
                </div>
        )
    }
}
