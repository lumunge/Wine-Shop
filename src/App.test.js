import React from 'react';
import Filter from './Components/Filter';
import {shallow} from 'enzyme'

describe('Filter Component Testing', () => {
    it('Should Render Correctly in Debug Mode', () => {
        const comp = shallow(<Filter debug/>);
        expect(comp).toMatchSnapshot();
    })
})


