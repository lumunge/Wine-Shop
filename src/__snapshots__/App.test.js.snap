// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Component Testing Should Render Correctly in Debug Mode 1`] = `ShallowWrapper {}`;
